src/features/create-dao/CreateDAOModal.tsx

```diff
@@ -0,0 +1,18 @@
+//- React Imports
+import { FC } from 'react';
+
+//- Component Imports
+import { Modal } from '@zero-tech/zui/components';
+import { CreateDAOForm } from './CreateDAOForm';
+
+//- Lib Imports
+import { BasicModalProps } from '../../lib/types/ui';
+
+export interface CreateDAOModalProps extends BasicModalProps {}
+
+// TODO - discuss with zer0 how to get domainName and manage close of modal.
```

Comment:
We can access `domain?.name` which is passed down from `App.tsx` (useDataContainer). So for `CreateDAOModal` you just need to add `domainName` to `CreateDAOModalProps`.

Let me know if that doesn't make sense. There are other examples of you search `domain?.name`.
https://github.com/zer0-os/zApp-NFTs/blob/7a2dfc8a0fe9f45e6c7c1761cf206ef6fdd20eaf/src/App.tsx#L25

Reply:
re onClose - for now I wouldn't worry too much about wiring it up as we're not exposing this modal for the time being. What you could do is add `onClose` to `CreateDAOModalProps` and pass it into `onClose` in place of `() => {}`. This is how I added it to `TransferOwnership`

https://github.com/zer0-os/zApp-NFTs/blob/7a2dfc8a0fe9f45e6c7c1761cf206ef6fdd20eaf/src/features/transfer-ownership/TransferOwnershipModal.tsx